relay/relay.go

```diff
@@ -129,14 +130,22 @@ func NewRelay(cfg *Config) *Relay {
 }
 
 // Init implements the dm.Unit interface.
-func (r *Relay) Init() error {
+func (r *Relay) Init() (err error) {
```

Comment:
Is it a little dangerous to use `err` in this way? For example if error happens in LINE#159, the rollback will not fire. We must ensure no `shadows declaration err` after `FuncRollback` is registered.

Reply:
`return` will **set** this `err`, then `defer` can detect it.

Reply:
well I see, it is the returned value